### DISCUSSION TITLE
NativeMenu stylization

### DISCUSSION THREAD

**Discussion by milkywayfarer** [AvaloniaUI/Avalonia#8301] [Q&A] [answered]
Searching the repository did not yield any results. Searching the `Avalonia.Themes.*/Controls` folder did not either.

How can one apply a style to the `NativeMenu`, `NativeMenuItem` and `NativeMenuItemSeparator`?

**Comment by maxkatz6** [comment, marked as answer]
It's native menu for a reason. You can't. 

Exception is when native menu is not supported or not implemented for specific backend platform. I.e. NativeMenuBar will be replaced with Menu control on Windows (with normal Menu styles), but not on mac or linux. And Windows TrayIcon will open avalonia popup with MenuItem items, but not on macos or linux where it will have true native menus.

**Reply by milkywayfarer** [reply]
And there are no other options to use rather than NativeMenu? I'm writing an only-Windows application using Avalonia.

**Reply by maxkatz6** [reply]
No, custom content in tray icon is not supported as for now.

**Reply by maxkatz6** [reply]
But if you to style native tray menu on windows, just style MenuItem control.

**Reply by milkywayfarer** [reply]
> But if you to style native tray menu on windows, just style MenuItem control.

Yes, that helped, thank you! Styled the Flyout and MenuItems.

**Reply by mingtong** [reply]
> No, custom content in tray icon is not supported as for now.

Is this possible in 2025 for now?
